### PR TITLE
Use the mutex pool provided by k8s keymutex

### PR DESCRIPTION
**Pull request by benashz**
The previous client cache locking scheme was not thread safe, and allocated more locks than are typically needed. This change replaces that approach by using the KeyMutex provided by the k8s utils package. Locks are now a pooled resource.

See https://pkg.go.dev/k8s.io/utils/keymutex#NewHashed for more info.

Other fixes
- update invalid Bitnami Helm chart repo for postgres. We should consider phasing out its use.
- Bump TF Helm to latest version

### CODE REVIEW

**Review by tvoran — APPROVED** [review]

File: test/unit/deployment.bats [hunk 1]
```diff
@@ -162,6 +162,61 @@ load _helpers
...
+  local object
+  object=$(helm template \
+      -s templates/deployment.yaml  \
+      --set 'controller.manager.clientCache.numLocks=-1' \
+      . | tee /dev/stderr |
+      yq 'select(.kind == "Deployment" and .metadata.labels."control-plane" == "controller-manager") | .spec.template.spec.containers[] | select(.name == "manager") | .args' | tee /dev/stderr)
+
+   local actual
+    actual=$(echo "$object" | yq 'contains(["--client-cache-num-locks=-1"])' | tee /dev/stderr)
+    [ "${actual}" = "true" ]
```
[the remark this answers is not in the document]
Comment (benashz) [reply test/unit/deployment.bats]: Fixed in 209d8a38c3cde06b6930d30305f0b379e742324e

File: test/unit/deployment.bats [hunk 2]
```diff
@@ -162,6 +162,61 @@ load _helpers
...
+  local object
+  object=$(helm template \
+      -s templates/deployment.yaml  \
+      --set 'controller.manager.clientCache.numLocks=0' \
+      . | tee /dev/stderr |
+      yq 'select(.kind == "Deployment" and .metadata.labels."control-plane" == "controller-manager") | .spec.template.spec.containers[] | select(.name == "manager") | .args' | tee /dev/stderr)
+
+   local actual
+    actual=$(echo "$object" | yq 'contains(["--client-cache-num-locks=0"])' | tee /dev/stderr)
+    [ "${actual}" = "true" ]
```
[the remark this answers is not in the document]
Comment (benashz) [reply test/unit/deployment.bats]: Fixed in 209d8a38c3cde06b6930d30305f0b379e742324e

File: test/unit/deployment.bats [hunk 3]
```diff
@@ -162,6 +162,61 @@ load _helpers
...
+  local object
+  object=$(helm template \
+      -s templates/deployment.yaml  \
+      --set 'controller.manager.clientCache.numLocks=500' \
+      . | tee /dev/stderr |
+      yq 'select(.kind == "Deployment" and .metadata.labels."control-plane" == "controller-manager") | .spec.template.spec.containers[] | select(.name == "manager") | .args' | tee /dev/stderr)
+
+   local actual
+    actual=$(echo "$object" | yq 'contains(["--client-cache-num-locks=500"])' | tee /dev/stderr)
+    [ "${actual}" = "true" ]
```
[the remark this answers is not in the document]
Comment (benashz) [reply test/unit/deployment.bats]: Fixed in 209d8a38c3cde06b6930d30305f0b379e742324e

File: chart/values.yaml [hunk 4]
```diff
@@ -267,6 +267,16 @@ controller:
...
 
+      # Defines the number of locks to use for the Vault client cache controller.
+      #
+      # Setting this value less than 1 will cause the manager to set the number of locks equal
+      # to the number of logical CPUs of the run host.
+      #
+      # See the VSO help output for more information.
+      #
+      # @type: integer
+      numLocks:
```
[the remark this answers is not in the document]
Comment (benashz) [reply chart/values.yaml:280]: Updated in dd5c52bd64db305d115a61d971502a8c715f7569

**Pull request merged**